src/main/scala/org/broadinstitute/hail/driver/GRM.scala

```diff
@@ -72,32 +75,70 @@ object GRM extends Command {
       && options.nFile != null)
       warn(s"format ${options.format}: ignoring `--N-file'")
 
+    val tmpFileName = state.hadoopConf.getTemporaryFile(HailConfiguration.tmpDir)
+
     options.format match {
       case "rel" =>
-        state.hadoopConf.writeTextFile(options.output) { s =>
-          for (i <- 0 until nSamples) {
-            for (j <- 0 to i) {
+        grm.toIndexedRowMatrix()
+          .rows
+          .sortBy(row => row.index)
```

Comment:
If this thing is large, sortBy is actually a VERY expensive operation.  It also requires computing the thing twice (once to sample ranges, once to shuffle)